Source/WebCore/platform/graphics/egl/GLContext.cpp

```diff
@@ -640,6 +640,25 @@ unsigned GLContext::version()
     return m_version;
 }
 
+GLContext::ScopedGLContext::ScopedGLContext(std::unique_ptr<GLContext>&& context)
+    : m_context(WTFMove(context))
+{
+    m_previous.context = eglGetCurrentContext();
+    if (m_previous.context) {
+        m_previous.display = eglGetCurrentDisplay();
+        m_previous.readSurface = eglGetCurrentSurface(EGL_READ);
+        m_previous.drawSurface = eglGetCurrentSurface(EGL_DRAW);
+    }
+    m_context->makeContextCurrent();
```

Comment:
The only reason where I can imagine this failing with GTK is if GTK doesn't use EGL but GLX.

Reply:
Right, in that case there'd be a current GLX context.

Calling `gdk_gl_context_clear_current()` would get rid of either, and GTK will call `make_current()` again, so calling that  is fine.
The other thing is that Webkit would need to make sure to call `clear_current()` (aka `eglMakeCurrent(NULL)`) once it's done so that GTK (if using GLX) doesn't cause problems when it calls make_current() later.

That reminds me: You might be able to reproduce this on X11 with `GDK_DEBUG=gl-glx` or `GDK_DEBUG=gl-egl` depending on what the problem here is - if it's the EGL/GLX interaction it should work, but if it requires a specific driver to cause that to fall over, then that won't be enough to trigger the fault.

Reply:
I'm not sure I understand... Does eglMakeCurrent affect a possible GLX current context? Note that we don't support GLX anymore in WebKit, so when GTK uses GLX (which is always for me under X11 because mesa doesn't provide any argb32 egl config) we fallback to cairo for rendering.

Reply:
Thanks for the explanation, it's clearer now. So, all this current context is per thread, right? I think that another possible solution might be to create and destroy the temporal EGL offscreen contexts in a separate thread. This is only used for webkit://gpu so, it should be ok to block the main thread while waiting for the EGL context information to be filled.